compiler/native/environment.go

```diff
@@ -109,6 +109,58 @@ func (c *client) EnvironmentServices(s yaml.ServiceSlice) (yaml.ServiceSlice, er
 	return s, nil
 }
 
+// EnvironmentSecrets injects environment variables
+// for each secret plugin in a yaml configuration.
+func (c *client) EnvironmentSecrets(s yaml.SecretSlice) (yaml.SecretSlice, error) {
+	// iterate through all secrets
+	for _, secret := range s {
+		// skip non plugin secrets
+		if secret.Origin.Empty() {
+			continue
+		}
+
+		// make empty map of environment variables
+		env := make(map[string]string)
+		// gather set of default environment variables
+		defaultEnv := environment(c.build, c.metadata, c.repo, c.user)
+
+		// inject the declared environment
+		// variables to the build secret
+		for k, v := range secret.Origin.Environment {
+			env[k] = v
+		}
+
+		// inject the default environment
+		// variables to the build secret
+		// we do this after injecting the declared environment
+		// to ensure the default env overrides any conflicts
+		for k, v := range defaultEnv {
+			env[k] = v
+		}
+
+		// inject the declared parameter
+		// variables to the build secret
+		for k, v := range secret.Origin.Parameters {
```

Comment:
it seems like this would be another place to override default variables, which https://github.com/go-vela/compiler/pull/76/files#diff-aaaa6f887aabe33da42651528fd46e70R137 is trying to prevent. should we move this block up before the mentioned one?

Reply:
@wass3r I don't believe we inject any `PARAMETER_*` environment variables by default 👍 

I think that is why this declaration is where it is.

Reply:
Fair :D